File: javascript/eliuds-eggs/eliuds-eggs.js
export const eggCount = (displayValue) => {
  let eggs = 0
  let max = 0
  while (2 ** max < displayValue) {
    max++
  }

  for (let i = max; i >= 0; i--) {
    if (displayValue >= 2 ** i) {
      eggs++
      displayValue -= 2 ** i
    }
  }

  return eggs
}

File: python/eliuds-eggs/eliuds_eggs.py
def egg_count(display_value):
    eggs = 0
    max = 0
    while 2 ** max < display_value:
        max += 1

    for i in range(max, -1, -1):
        if display_value >= 2 ** i:
            eggs += 1
            display_value -= 2 ** i

    return eggs
